packages/flutter_tools/test/web.shard/chrome_test.dart

```diff
@@ -821,6 +821,49 @@ void main() {
     expect(commands, contains('Browser.close'));
   });
 
+  testWithoutContext('chrome.close can recover if getTab throws an HttpException', () async {
```

Comment:
Do these tests verify the change you're making? I tried reverting your change to packages/flutter_tools/lib/src/web/chrome.dart, and it seems like these tests still pass. LMK if that's unexpected, I may have messed up my git foo.

Reply:
Yes, this is unexpected. I think I can fix this.

Reply:
Coming back to this, the test you highlighted here (`chrome.close can recover if getTab throws an HttpException`) doesn't verify this change because it doesn't throw the exception from another zone. I personally don't know how to simulate this, so I attempted to simply remove this test.

The other test (`chrome.close can recover if getTab throws a StateError`) was indeed non-functional and fixed in master by https://github.com/flutter/flutter/pull/154889. I've copied the changes to this PR.